src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,122 @@
+                manufacturer = new Manufacturer();
+                parseResultSet(resultSet, manufacturer);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all manufacturers from DB", e);
+        }
+        return Optional.ofNullable(manufacturer);
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String getAllManufacturersRequest = "SELECT * FROM manufacturers WHERE is_deleted = false;";
+        List<Manufacturer> allManufacturers = new ArrayList<>();
+        try (Connection connection = connectionDB.getConnect();
+                 PreparedStatement getAllManufacturers = connection
+                         .prepareStatement(getAllManufacturersRequest)) {
+            ResultSet resultSet = getAllManufacturers.executeQuery();
+            while (resultSet.next()) {
+                Manufacturer manufacturer = new Manufacturer();
+                allManufacturers.add(parseResultSet(resultSet, manufacturer));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all manufacturers from DB", e);
+        }
+        return allManufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateManufactureRequest = "UPDATE manufacturers SET name = ?, country = ? "
+                + "WHERE is_deleted = false AND id  = ?;";
+        try (Connection connection = connectionDB.getConnect();
+                 PreparedStatement updateManufactureStatement =
+                         connection.prepareStatement(updateManufactureRequest)) {
+            updateManufactureStatement.setString(1, manufacturer.getName());
+            updateManufactureStatement.setString(2, manufacturer.getCountry());
+            updateManufactureStatement.setLong(3, manufacturer.getId());
+            Manufacturer oldManufacturer = get(manufacturer.getId()).orElseThrow(SQLException::new);
+            updateManufactureStatement.executeUpdate();
+            return oldManufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update manufacturer by id "
+                    + manufacturer.getId() + " from DB", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE manufacturers SET is_deleted = true WHERE id  = ?;";
+        try (Connection connection = connectionDB.getConnect();
+                 PreparedStatement deleteStatement =
+                         connection.prepareStatement(deleteRequest)) {
+            deleteStatement.setLong(1, id);
+            return deleteStatement.executeUpdate() >= 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete manufacturer by id "
+                    + id + " from DB", e);
+        }
+    }
+
+    private Manufacturer parseResultSet(ResultSet resultSet, Manufacturer manufacturer)
```

Comment:
```suggestion
    private Manufacturer parseResultSet(ResultSet resultSet, Manufacturer manufacturer)
```
Мне кажется можно не передавать объект Manufacturer , а создавать его уже в этом методе.